### ISSUE TITLE
minikube (qemu2) failes to start with "failed to get IP address"

### ISSUE THREAD

**Issue by denisko** [kubernetes/minikube#15960]
### What Happened?

Minikube fails to start:
`Exiting due to GUEST_PROVISION: Failed to start host: creating host: create: creating: IP address never found in dhcp leases file: failed to get IP address: could not find an IP address for xx:xx:xx:xx:xx:xx`

### Attach the log file

[minikube-logs.txt](https://github.com/kubernetes/minikube/files/10880333/minikube-logs.txt)
[minikube_logs_e4032c57df89461a76cf8ea248c98ea6558dccb1_0.log](https://github.com/kubernetes/minikube/files/10881561/minikube_logs_e4032c57df89461a76cf8ea248c98ea6558dccb1_0.log)

### Operating System

macOS (Default)

### Driver

QEMU

**Comment by afbjorklund** [comment]
Some kind of compatibility issue with socket_vmnet, it only lists one DHCP entry coming from "colima"

```
I0303 10:05:06.492396   25101 main.go:141] libmachine: Searching for 2:a8:48:27:20:f2 in /var/db/dhcpd_leases ...
I0303 10:05:06.492450   25101 main.go:141] libmachine: Found 1 entries in /var/db/dhcpd_leases!
I0303 10:05:06.492474   25101 main.go:141] libmachine: dhcp entry: {Name:colima IPAddress:192.168.106.2 HWAddress:52:55:55:da:34:59 ID:1,52:55:55:da:34:59 Lease:0x63eb6dd7}
```

**Comment by mprimeaux** [comment]
I'm not sure what is going on yet but I am also experiencing this issue, which **seems** to be a regression as compared to versions before minikube `1.30.x` **but** I do need to verify. 

I'll post logs but, in summary, I'm unable to successfully start a minikube cluster when using `socket_vmnet` (via `brew` or direct source compilation) in NAT or bridged mode. This used to work but does not even after following steps outlined in [Known Issues](https://minikube.sigs.k8s.io/docs/drivers/qemu/#known-issues).

Here are my machine details:

```shell
❯ macchina                                                                                                                                                 ─╯

                  ,MMMM.           Host        -  <REMOVED>
                .MMMMMM            Machine     -  Mac13,2
                MMMMM,             Kernel      -  22.4.0
      .;MMMMM:' MMMMMMMMMM;.       OS          -  macOS 13.3.1 Ventura
    MMMMMMMMMMMMNWMMMMMMMMMMM:     DE          -  Aqua
  .MMMMMMMMMMMMMMMMMMMMMMMMWM.     WM          -  Quartz Compositor
  MMMMMMMMMMMMMMMMMMMMMMMMM.       Packages    -  263 (Homebrew)
 ;MMMMMMMMMMMMMMMMMMMMMMMM:        Terminal    -  iTerm2 (Version 3.4.19)
 :MMMMMMMMMMMMMMMMMMMMMMMM:        Shell       -  zsh
 .MMMMMMMMMMMMMMMMMMMMMMMMM.       Uptime      -  3d 5h 49m
  MMMMMMMMMMMMMMMMMMMMMMMMMMM.     CPU         -  Apple M1 Ultra (20)
   .MMMMMMMMMMMMMMMMMMMMMMMMMM.    Resolution  -  5120x2880@60fps (as 2560x1440)
     MMMMMMMMMMMMMMMMMMMMMMMM                     5120x2880@60fps (as 2560x1440)
      ;MMMMMMMMMMMMMMMMMMMM.       Brightness  -  100%
        .MMMM,.    .MMMM,.         CPU Load    -  18%
                                   Memory      -  60.2 GB/134.2 GB
```

Minikube version is:

```shell
minikube version: v1.30.1
commit: 08896fd1dc362c097c925146c4a0d0dac715ace0
```

I will do more testing tomorrow.

**Comment by mprimeaux** [comment]
FWIW, here are my logs from today's latest attempt to start minikube with `socket_vmnet`. 

[minikube_logs_246a8c5e046f473e09b808593858b4737e8d9296_0.log](https://github.com/kubernetes/minikube/files/11305102/minikube_logs_246a8c5e046f473e09b808593858b4737e8d9296_0.log)

```shell
❯ minikube config view
- container-runtime: containerd
- cpus: 6
- driver: qemu2
- memory: 16384
```

```shell
❯ minikube start --network socket_vmnet
😄  minikube v1.30.1 on Darwin 13.3.1 (arm64)
✨  Using the qemu2 driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
💾  Downloading Kubernetes v1.26.3 preload ...
    > preloaded-images-k8s-v18-v1...:  358.98 MiB / 358.98 MiB  100.00% 44.94 M
🔥  Creating qemu2 VM (CPUs=6, Memory=16384MB, Disk=20000MB) ...
🔥  Deleting "minikube" in qemu2 ...
🤦  StartHost failed, but will try again: creating host: create: creating: IP address never found in dhcp leases file: failed to get IP address: could not find an IP address for 5e:40:c0:5c:ab:86
🔥  Creating qemu2 VM (CPUs=6, Memory=16384MB, Disk=20000MB) ...
😿  Failed to start qemu2 VM. Running "minikube delete" may fix it: creating host: create: creating: IP address never found in dhcp leases file: failed to get IP address: could not find an IP address for 42:74:86:6:b8:92

❌  Exiting due to GUEST_PROVISION: error provisioning guest: Failed to start host: creating host: create: creating: IP address never found in dhcp leases file: failed to get IP address: could not find an IP address for 42:74:86:6:b8:92

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯
```

I'm using `socket_vmnet` via `brew` but not in bridged mode. 

```shell
❯ sudo brew services info socket_vmnet 
socket_vmnet (homebrew.mxcl.socket_vmnet)
Running: ✔
Loaded: ✔
Schedulable: ✘
User: root
PID: 743
```

I've also tried different values of `container-runtime` to no avail. I'll find more time tomorrow to debug a bit deeper.

**Comment by torenware** [comment]
Just tried this with newly released minikube v1.31.0.  Still busted on amd64 (might be working as 1.30.x on arm64!)

**Comment by mprimeaux** [comment]
Still busted on arm64, also. As several folks have mentioned, the fix for me on both amd64 and arm64 is to restart socket_vmnet via `brew services` and then to run the following:

```shell
 Ref: https://github.com/lima-vm/socket_vmnet/issues/18#issuecomment-1574149506
sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd
sudo /usr/libexec/ApplicationFirewall/socketfilterfw --unblock /usr/libexec/bootpd
```

This workaround works for me 100% of the time.

**Comment by spowelljr** [comment]
minikube v1.31.0 should unblock the firewall if it detects it's being blocked.

```
$ $ ./out/minikube start --driver qemu
😄  minikube v1.31.0 on Darwin 13.4.1 (arm64)
✨  Using the qemu2 driver based on user configuration
🌐  Automatically selected the socket_vmnet network
🔑  Your firewall is blocking bootpd which is required for socket_vmnet. The following commands will be executed to unblock bootpd:

    $ sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd 
    $ sudo /usr/libexec/ApplicationFirewall/socketfilterfw --unblock /usr/libexec/bootpd 


Password: 
👍  Starting control plane node minikube in cluster minikube
💾  Downloading Kubernetes v1.27.3 preload ...
    > preloaded-images-k8s-v18-v1...:  327.72 MiB / 327.72 MiB  100.00% 50.68 M
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
❗  This VM is having trouble accessing https://registry.k8s.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🐳  Preparing Kubernetes v1.27.3 on Docker 24.0.4 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

**Comment by torenware** [comment]
> Still busted on arm64, also. As several folks have mentioned, the fix for me on both amd64 and arm64 is to restart socket_vmnet via `brew services` and then to run the following:
> 
> ```shell
> minikube delete
> sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd
> sudo /usr/libexec/ApplicationFirewall/socketfilterfw --unblock /usr/libexec/bootpd
> ```
> 
> This workaround works for me 100% of the time.

I can't get it to run on amd64, even with the ownership on ` /usr/local/var/run/socket_vmnet` set to `root:admin` (which makes it possible to run `minikube start` as non-root).  Here's what I see on my Intel Macbook Pro, immediately after giving the user password for sudo:

```
Robs-MBP:images myuser$ minikube -p vmnet delete
🔥  Deleting "vmnet" in qemu2 ...
💀  Removed all traces of the "vmnet" cluster.
Robs-MBP:images myuser$ minikube -p vmnet start --driver qemu
😄  [vmnet] minikube v1.31.0 on Darwin 11.7.8
✨  Using the qemu2 driver based on user configuration
🌐  Automatically selected the socket_vmnet network
👍  Starting control plane node vmnet in cluster vmnet
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
🔑  Your firewall is blocking bootpd which is required for socket_vmnet. The following commands will be executed to unblock bootpd:

    $ sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd
    $ sudo /usr/libexec/ApplicationFirewall/socketfilterfw --unblock /usr/libexec/bootpd


🔄  Successfully unblocked bootpd process from firewall, retrying
🔥  Deleting "vmnet" in qemu2 ...
🤦  StartHost failed, but will try again: creating host: create: creating: ip not found: failed to get IP address: could not find an IP address for 32:82:30:a9:ad:e1
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
🔑  Your firewall is blocking bootpd which is required for socket_vmnet. The following commands will be executed to unblock bootpd:

    $ sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd
    $ sudo /usr/libexec/ApplicationFirewall/socketfilterfw --unblock /usr/libexec/bootpd


🔄  Successfully unblocked bootpd process from firewall, retrying
😿  Failed to start qemu2 VM. Running "minikube delete -p vmnet" may fix it: creating host: create: creating: ip not found: failed to get IP address: could not find an IP address for b2:f9:59:88:37:d6

❌  Exiting due to GUEST_PROVISION: error provisioning guest: Failed to start host: creating host: create: creating: ip not found: failed to get IP address: could not find an IP address for b2:f9:59:88:37:d6

╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯

```

**Comment by torenware** [comment]
[minikube.log](https://github.com/kubernetes/minikube/files/12090043/minikube.log)

**Comment by mprimeaux** [comment]
Did you run `sudo brew services restart socket_vmnet` prior to the above?

**Comment by torenware** [comment]
@mprimeaux Yep.  Here's the status:
```
Robs-MBP:images myuser$ sudo brew services info socket_vmnet
socket_vmnet (homebrew.mxcl.socket_vmnet)
Running: ✔
Loaded: ✔
Schedulable: ✘
User: root
PID: 94341
```

**Comment by mprimeaux** [comment]
@torenware There are days I wish I were on a beach :)

**Comment by torenware** [comment]
File:  

> @torenware There are days I wish I were on a beach :)
> 
> Interestingly enough, I had this very scenario happen when I was in an airport last week. Turns our, the captured network was preventing my VM from obtaining a DHCP address. By chance, are you on a captured network?

Nope. Plain vanilla wifi, no fancy stuff going on.

**Comment by torenware** [comment]
[logs.txt](https://github.com/kubernetes/minikube/files/12090096/logs.txt)

**Comment by richpryce** [comment]
I am having this exact same issue as well. Has a fix been found?

**Comment by afbjorklund** [comment]
It was not intended to start two VMs, so it seems strange to have both colima and minikube qemu2 running.

Overriding the user name is something that could be considered for all drivers, but it is not needed for the kic image.

**Comment by obnoxxx** [comment]
@mprimeaux  wrote: 
> Still busted on arm64, also. As several folks have mentioned, the fix for me on both amd64 and arm64 is to restart socket_vmnet via `brew services` and then to run the following:
> 
> ```shell
> minikube delete
> sudo /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd
> sudo /usr/libexec/ApplicationFirewall/socketfilterfw --unblock /usr/libexec/bootpd
> ```
> 
> This workaround works for me 100% of the time.

This worked for me once yesterday, and I was so happy, but today it fails again:

```console
GUEST_PROVISION: error provisioning guest: Failed to start host: creating host: create: creating: ip not found: failed to get IP address: could not find an IP address for 66:17:20:72:c8:65

```

In my case, the firewall does *not* seem to be the problem because `bootpd`is listed there. But my suspicion is that the problem is that `bootpd`is not even running. How can I get it to run?

**Comment by obnoxxx** [comment]
I managed to get bootpd running by doing `sudo /usr/libexec/bootpd` but it doesnt help minikube starting. Still failing with `failed to get IP address`

**Comment by avoidik** [comment]
not sure why but it does allocate ipv6 ip address instead of expected ipv4 (a primary interface is eth1 just because eth0 is used for troubleshooting):

```
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 66:49:bf:12:9a:f9 brd ff:ff:ff:ff:ff:ff
    inet6 fde9:a9ca:323b:9437:6449:bfff:fe12:9af9/64 scope global dynamic mngtmpaddr noprefixroute
       valid_lft 2591985sec preferred_lft 604785sec
    inet6 fe80::6449:bfff:fe12:9af9/64 scope link
       valid_lft forever preferred_lft forever
```

then, using ipv6 ip address I am able to connect over ssh

```terminal
$ ssh -i ~/.minikube/machines/minikube/id_rsa docker@fde9:a9ca:323b:9437:6449:bfff:fe12:9af9 whoami
Warning: Permanently added 'fde9:a9ca:323b:9437:6449:bfff:fe12:9af9' (ED25519) to the list of known hosts.
docker
```
